src/DataAPI.jl

```diff
@@ -69,6 +69,26 @@ default definition.
 function refpool end
 refpool(A::AbstractArray) = nothing
 
+"""
+    invrefpool(A)
+
+Whenever available, return an indexable object such that given an array `A` for which `refpool(A)` is not `nothing` such that:
+
+* for any valid index `x` into `refpool(A)` the following holds `invrefpool(A)[refpool(A)[x]] === x`;
+* for any valid index `ix` into `invrefpool(A)` the following holds `refpool(A)[invrefpool(A)[ix]] === ix`.
+
+Additionally it is requred that `haskey` is defined for `invrefpool(A)` allowing to check if `ix` is a valid index into it.
```

Comment:
Currently we don't require `=== ` in the docstring for `refpool` (only `isequal` and equal type). We could discuss changing this if we have a reason to, but I'm not sure we have.
```suggestion
Whenever available, return an indexable object such that given an array `A`
for which `refpool(A)` is not `nothing`:

* for any valid index `x` into `refpool(A)`, `invrefpool(A)[refpool(A)[x]]` is equal to `x`
  (according to `isequal`) and of the same type as `x`;
* for any valid index `ix` into `invrefpool(A)` , `refpool(A)[invrefpool(A)[ix]]` is equal to `ix`
  (according to `isequal`) and of the same type as `ix`.

Additionally it is required that `haskey` is defined for `invrefpool(A)`,
allowing to check if `ix` is a valid index into it.
```